Sources/Sentry/SentryScope.m

```diff
@@ -251,142 +319,132 @@ - (void)setLevel:(enum SentryLevel)level
-                [newTags addEntriesFromDictionary:event.tags];
-                event.tags = newTags;
-            }
-        }
-
-        if (nil != self.extraDictionary) {
-            if (nil == event.extra) {
-                event.extra = self.extraDictionary.copy;
-            } else {
-                NSMutableDictionary *newExtra = [NSMutableDictionary new];
-                [newExtra addEntriesFromDictionary:self.extraDictionary];
-                [newExtra addEntriesFromDictionary:event.extra];
-                event.extra = newExtra;
-            }
-        }
+    if (nil == event.tags) {
+        event.tags = [self tags];
+    } else {
+        NSMutableDictionary *newTags = [NSMutableDictionary new];
+        [newTags addEntriesFromDictionary:[self tags]];
+        [newTags addEntriesFromDictionary:event.tags];
+        event.tags = newTags;
+    }
 
-        if (nil != self.userObject) {
-            event.user = self.userObject.copy;
-        }
+    if (nil == event.extra) {
+        event.extra = [self extras];
+    } else {
+        NSMutableDictionary *newExtra = [NSMutableDictionary new];
+        [newExtra addEntriesFromDictionary:[self extras]];
+        [newExtra addEntriesFromDictionary:event.extra];
+        event.extra = newExtra;
+    }
 
-        NSString *dist = self.distString;
-        if (nil != dist && nil == event.dist) {
-            // dist can also be set via options but scope takes precedence.
-            event.dist = dist;
-        }
+    NSArray *fingerprints = [self fingerprints];
+    if (fingerprints.count > 0 && nil == event.fingerprint) {
+        event.fingerprint = fingerprints;
+    }
 
-        NSString *environment = self.environmentString;
-        if (nil != environment && nil == event.environment) {
-            // environment can also be set via options but scope takes
-            // precedence.
-            event.environment = environment;
-        }
+    if (nil == event.breadcrumbs) {
+        NSArray *breadcrumbs = [self breadcrumbs];
+        event.breadcrumbs = [breadcrumbs
+            subarrayWithRange:NSMakeRange(0, MIN(maxBreadcrumbs, [breadcrumbs count]))];
+    }
 
-        NSArray *fingerprint = self.fingerprintArray;
-        if (fingerprint.count > 0 && nil == event.fingerprint) {
-            event.fingerprint = fingerprint.mutableCopy;
-        }
+    if (nil == event.context) {
+        event.context = [self context];
+    } else {
+        NSMutableDictionary *newContext = [NSMutableDictionary new];
+        [newContext addEntriesFromDictionary:[self context]];
+        [newContext addEntriesFromDictionary:event.context];
+        event.context = newContext;
+    }
 
-        if (self.levelEnum != kSentryLevelNone) {
-            // We always want to set the level from the scope since this has
-            // benn set on purpose
-            event.level = self.levelEnum;
-        }
+    if (nil != self.userObject) {
+        event.user = self.userObject.copy;
```

Comment:
`userObject` could be `nil` on 424 without synchronizing access to `userObject` or having a copy of the reference. Copying the reference still means its state could change after the fact. Do we have `other` on User here? It's a map usually, so we'd need to synchronize that too

Reply:
Now we copy the `userObject` to a local variable. `SentryUser` implements `copyWithZone` so if it would change afterward, we are safe.